backend/python-functions/data_ingest.py

```diff
@@ -1,12 +1,102 @@
 import csv
+from datetime import datetime, timezone
 from google.cloud import firestore
 from google.cloud import storage
 import logging
+import re
 
 REFERENCE_FACILITIES_COLLECTION_ID = 'reference_facilities'
+# this limit is imposed by firestore
+BATCH_SIZE = 500
 
 log = logging.getLogger("cloudLogger")
 
+# translate bigquery values to the ones we use in this app
+FACILITY_TYPE_MAPPING = {
+    "County Jails": "County Jail",
+    "State Prisons": "State Prison",
+}
+
+fs_client = firestore.Client()
+facilities_collection = fs_client.collection(
+    REFERENCE_FACILITIES_COLLECTION_ID)
+
+def create_or_update_facilities(file_location):
+    batch = fs_client.batch()
+    with open(file_location, newline='') as f:
+        reader = csv.DictReader(f)
+
+        batch_counter = 0
+        for row in reader:
+            # what appears to be a single "write" can contain multiple operations;
+            # we need to keep track of them to stay below the Firestore batch limit
+            row_ops_count = 1
+            id = row['facility_id']
+
+            facility_doc_ref = facilities_collection.document(id)
+            facility_dict = {}
+
+            # are we updating or creating?
+            try:
+                fdoc_snapshot = facility_doc_ref.get()
+                assert fdoc_snapshot.exists
```

Comment:
Personally, I have an aversion to using exceptions to conduct normal, accepted control flows. I would just do a standard if-else check on `.exists` here instead of asserting.